### PR TITLE
Add support for Lippert LP sensors in mopeka_pro_check component

### LINKED ISSUE #1988: Add support for Lippert Propane Sensors
**Describe the problem you have/What new integration you would like**
The current Mopeka_pro_check component does not support Lippert propane sensors even though they are built on the same hardware platform.

**Please describe your use case for this integration and alternatives you've tried:**
The Lippert sensors report the same data, but a different manufacturer ID to the Mopeka component.  A minor change in the component will support this (PR to come)

**Additional context**

### PR DESCRIPTION
**Pull request by rperciaccante**
# What does this implement/fix?

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes [<link to issue>](https://github.com/esphome/feature-requests/issues/1988)

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#2493

## Test Environment

- [X] ESP32
- [X] ESP32 IDF
- [X] ESP8266
- [X] RP2040

## Example entry for `config.yaml`:

```yaml
#No changes from existing YAML configuration: 
esp32_ble_tracker:

sensor:
  # Example using 20lb vertical propane tank.
  - platform: mopeka_pro_check
    mac_address: D3:75:F2:DC:16:91
    tank_type: 20LB_V
    temperature:
        name: "Propane test temp"
    level:
        name: "Propane test level"
    distance:
        name: "Propane test distance"
    battery_level:
        name: "Propane test battery level"

```

## Checklist:
  - [X] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Review by spbrogan — APPROVED** [review]
Thanks for adding support for a new model of sensor.  
You should probably add a documentation update here: https://github.com/esphome/esphome-docs/blob/current/components/sensor/mopeka_pro_check.rst  
to indicate support for the new sensor model.

**Comment by rperciaccante** [comment]
> Thanks for adding support for a new model of sensor. You should probably add a documentation update here: https://github.com/esphome/esphome-docs/blob/current/components/sensor/mopeka_pro_check.rst to indicate support for the new sensor model.

I submitted a new PR for that as well - https://github.com/esphome/esphome-docs/pull/2493

File: esphome/components/mopeka_pro_check/mopeka_pro_check.cpp [hunk 1]
```diff
@@ -50,14 +50,21 @@ bool MopekaProCheck::parse_device(const esp32_ble_tracker::ESPBTDevice &device)
     return false;
   }
 
-  // Now parse the data - See Datasheet for definition
+  // Check to see if the manufacturer is supported
+  //if (static_cast < SensorType > (manu_data.data[0]) != STANDARD_BOTTOM_UP && static_cast < SensorType > (manu_data.data[0]) != LIPPERT_BOTTOM_UP && static_cast < SensorType > (manu_data.data[0]) != PLUS_BOTTOM_UP) {
+  //  ESP_LOGE(TAG, "Unsupported Sensor Type (0x%X)", manu_data.data[0]);
+  //  return false;
+  //}
+
```
Comment (jesserockz) [inline esphome/components/mopeka_pro_check/mopeka_pro_check.cpp]: No commented code please
```suggestion
```

File: esphome/components/mopeka_pro_check/mopeka_pro_check.cpp [hunk 2]
```diff
@@ -50,14 +50,21 @@ bool MopekaProCheck::parse_device(const esp32_ble_tracker::ESPBTDevice &device)
...
 
   if (static_cast<SensorType>(manu_data.data[0]) != STANDARD_BOTTOM_UP &&
-      static_cast<SensorType>(manu_data.data[0]) != PLUS_BOTTOM_UP) {
+      static_cast<SensorType>(manu_data.data[0]) != LIPPERT_BOTTOM_UP &&
+      static_cast<SensorType>(manu_data.data[0]) != PLUS_BOTTOM_UP ) {
```
Comment (jesserockz) [inline esphome/components/mopeka_pro_check/mopeka_pro_check.cpp]:
```suggestion
      static_cast<SensorType>(manu_data.data[0]) != PLUS_BOTTOM_UP) {
```

File: esphome/components/mopeka_pro_check/mopeka_pro_check.cpp [hunk 3]
```diff
@@ -50,14 +50,21 @@ bool MopekaProCheck::parse_device(const esp32_ble_tracker::ESPBTDevice &device)
...
     ESP_LOGE(TAG, "Unsupported Sensor Type (0x%X)", manu_data.data[0]);
     return false;
   }
 
+
```
Comment (jesserockz) [inline esphome/components/mopeka_pro_check/mopeka_pro_check.cpp]: Removing extra blank line
```suggestion
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: esphome/components/mopeka_pro_check/mopeka_pro_check.cpp [hunk 4]
```diff
@@ -50,9 +50,9 @@ bool MopekaProCheck::parse_device(const esp32_ble_tracker::ESPBTDevice &device)
     return false;
   }
 
-  // Now parse the data - See Datasheet for definition
```
Comment (jesserockz) [inline esphome/components/mopeka_pro_check/mopeka_pro_check.cpp:54]: Add back original comment
```suggestion
// Now parse the data - See Datasheet for definition

```

**Review by rperciaccante — COMMENTED** [review]
Thank you - I think it is ready to go - please let me know if I am missing anything else

**Comment by spbrogan** [comment]
Now that all "checks" are passing the content looks good.  Hopefully since this is such a small but important change (adding support for an additional sensor) this can get merged soon.   Thanks again.

**Comment by rperciaccante** [comment]
@jesserockz was there something that I missed that prevents this from being merged?

**Comment by jesserockz** [comment]
Sorry, just been too busy...haha

**Review by jesserockz — APPROVED** [review]

**Pull request merged**